e2e/testcases/custom_resources_test.go

```diff
@@ -109,16 +109,13 @@ func TestCRDDeleteBeforeRemoveCustomResourceV1(t *testing.T) {
 		nt.T.Fatal(err)
 	}
 
-	// Wait for remediator to detect the drift (managed Anvil CR was deleted)
-	// and record it as a conflict.
-	// TODO: distinguish between management conflict (spec/generation drift) and concurrent status update conflict (resource version change)
-	err = nomostest.ValidateMetrics(nt,
+	// Reverting a manual change is NOT considered a "conflict".
+	// Conflict implies either a different reconciler manager or drift between GET (or watch event) & UPDATE.
+	// TODO: distinguish between management conflict (unexpected manager annotation) and resource conflict (resource version change)
+	nt.Must(nomostest.ValidateMetrics(nt,
 		nomostest.ReconcilerErrorMetrics(nt, rootSyncLabels, firstCommitHash, metrics.ErrorSummary{
-			Conflicts: 1,
-		}))
-	if err != nil {
-		nt.T.Fatal(err)
-	}
+			Conflicts: 0,
```

Comment:
The intent of the check here is to confirm the remediator detects the drift, but couldn't correct it.

Instead of checking no conflicts, what about adding a new metric for the [remediator error](https://github.com/GoogleContainerTools/kpt-config-sync/blob/v1.18.3-rc.4/pkg/remediator/reconcile/reconciler.go#L94), and validating a remediator error is recorded in the metrics? We can still check no conflicts, but it is just no enough.

Reply:
I added handling of NoMatchErrors as conflicts using the existing metric. This should make it report the error metric consistently in this test.

If we want to add a new metric, it'll need a one-pager, because we also need to separate management conflicts from resource conflict, and list all the cases to make sure they're handled correctly. The management conflict reporting is spotty at best today. It only works for RootSyncs and only sometimes.  I don't really have time to do this now, due to the watch filter project.